eng/pipelines/scripts/Get-Linting-Commands.ps1

```diff
@@ -0,0 +1,80 @@
+<#
+.SYNOPSIS
+Determines the Checkstyle, RevApi, and Spotbugs linting commands that should be ran as validation in the
+code-quality-reports pipeline.
+
+.DESCRIPTION
+Given a build type and target branch this will determine which Checkstyle, RevApi, and Spotbugs linting commands
+will be used as validation in the code-quality-reports pipeline.
+
+If the build type is "Scheduled" all linting steps will be performed as this is a daily validation. This will verify
+all code changes since the last scheduled build against Checkstyle, RevApi, and Spotbugs linting.
+
+Otherwise, the linting steps are determined based on which directories have code modifications. They are the following:
+
+* checkstyle:check - eng/code-quality-reports/src/main/java/* (indicates source code changes to our custom Checkstyle rules)
+* checkstyle:check - eng/code-quality-reports/src/main/resources/checkstyle/* (indicates Checkstyle configuration changes)
+* revapi:check - eng/code-quality-reports/src/main/resources/revapi/* (indicates RevApi configuration changes)
+* spotbugs:check - eng/code-quality-reports/src/main/resources/spotbugs/* (indicates Spotbugs configuration changes)
+
+.PARAMETER BuildReason
+Which pipeline build reason triggered execution of code-quality-reports.
+
+.PARAMETER SourceBranch
+The branch containing changes.
+
+.PARAMETER TargetBranch
+The branch being merged into once CI passes.
+
+.PARAMETER LintingPipelineVariable
+The pipeline variable that should be set containing the linting goals.
+#>
+
+param(
+  [Parameter(Mandatory = $true)]
+  [string]$BuildReason,
+
+  [Parameter(Mandatory = $true)]
+  [string]$SourceBranch,
+
+  [Parameter(Mandatory = $true)]
+  [string]$TargetBranch,
+
+  [Parameter(Mandatory = $true)]
+  [string]$LintingPipelineVariable
+)
+
+Write-Host "Build reason: ${BuildReason}"
+Write-Host "Source branch: ${SourceBranch}"
+Write-Host "Target branch: ${TargetBranch}"
+Write-Host "Linting pipeline variable: ${LintingPipelineVariable}"
+
+if ($BuildReason -eq "Scheduled") {
```

Comment:
I believe you only want to do the bottom checks for pull requests so perhaps you invert this condition to say not PullRequest.

Reply:
At this time, I only wanted it to run the full suite of linting when it is a scheduled build but I agree this should also be updated for other potential cases such as a one-off manual build against main.